service/src/main/java/com/epam/esm/service/UserService.java

```diff
@@ -106,20 +127,34 @@ public User getUserById(int userId) {
     }
 
     public Tag getMostUsedTagOfUserWithHighestCostOfAllOrders() {
-        Optional<Tag> tag = userRepository.findMostUsedTagOfUserWithHighestCostOfAllOrders();
+        Optional<Tag> tag = tagRepository.findMostUsedTagOfUserWithHighestCostOfAllOrders();
         if (!tag.isPresent()) {
             throw new ServiceException("No orders in database");
         }
         return tag.get();
     }
 
     public PageInfo userPageInfo(int pageNumber, int pageSize) {
-        int usersCount = userRepository.countAll();
+        int usersCount = (int)userRepository.count();
         return new PageInfo(pageSize, pageNumber, usersCount);
     }
 
     public PageInfo userOrdersPageInfo(int userId, int pageNumber, int pageSize) {
-        int userOrdersCount  = userOrderUtil.countUserOrders(userId);
+        int userOrdersCount  = orderRepository.countUserOrders(userId);
         return new PageInfo(pageSize, pageNumber, userOrdersCount);
     }
+
+    public User register(User user) {
+        Optional<User> userOptional = userRepository.findByUsername(user.getUsername());
+        if (userOptional.isPresent()) {
+            throw new ServiceException(USER_WITH_SUCH_USERNAME_EXISTS);
+        }
+        user.setBalance(BigDecimal.ZERO);
+        user.setOrders(null);
```

Comment:
I suppose it is better to use empty list here. What do you think?

Reply:
I think there is no difference. JPA will replace it with its own proxy list, but I've changed id just in case